pontoon/base/migrations/0018_populate_entity_context.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 3.2.4 on 2021-08-19 13:46
+
+from django.db import migrations
+
+from pontoon.sync import KEY_SEPARATOR
+
+
+def add_entity_context(apps, schema_editor):
+    Entity = apps.get_model("base", "Entity")
+    entities = Entity.objects.all().prefetch_related("resource")
+
+    for e in entities:
+        if e.resource.format == "lang":
+            e.context = ""
+        if e.resource.format == "po":
+            e.context = e.key.split(KEY_SEPARATOR)[0]
```

Comment:
*idea* Postgresql has `split_part` function, maybe something like this could be a little bit faster (in terms of the time and memory):
```
 split_key_po  = Func(F('key'), Value(KEY_SEPARATOR), Value(1), function='split_part', output_field=TextField())
 split_key_xliff = Func(F('key'), Value(KEY_SEPARATOR), Value(2), function='split_part', output_field=TextField())
Entity.objects.filter(resource__format='lang').update(context='')
Entity.objects.filter(resource__format='po').update(context=split_key_po)
Entity.objects.filter(resource__format='xliff').update(context=split_key_xliff)
Entity.objects.exclude(resource__format__in=['lang', 'po', 'xliff']).update(context=F('key'))
```
You can also try to create a raw SQL to make everything in one query:
```
UPDATE base_entity SET context=CASE
WHEN format='po' THEN split_part(key, '<KEY_SEPARATOR> ', 1)
WHEN format='xliff' THEN split_part(string, '<KEY_SEPARATOR> ', 2)
WHEN format='lang' THEN ''
ELSE key
END
FROM base_resource WHERE base_entity.resource_id=base_resource.id
```

Reply:
Thanks, I'll update!

This is a one-off data migration and it only took about a minute to complete on stage with prod data, but I expect it to be much faster with your approach. Maybe someone else will benefit even more than our instance.